entitybroker/utils/src/java/org/sakaiproject/entitybroker/util/servlet/DirectServlet.java

```diff
@@ -147,12 +147,13 @@ protected void handleRequest(HttpServletRequest req, HttpServletResponse res)
             throws ServletException, IOException {
         // catch the login helper posts
         String option = req.getPathInfo();
-        String[] parts = option.split("/");
-        if ((parts.length == 2) && ((parts[1].equals("login")))) {
-            handleUserLogin(req, res, null);
-        } else {
-            dispatch(req, res);
+        if (option != null) {
+            final String[] parts = option.split("/");
+            if ((parts.length == 2) && ((parts[1].equals("login")))) {
+                handleUserLogin(req, res, null);
+            }
         }
+        dispatch(req, res);
```

Comment:
Won't it be the case that when the login tool has rendered the page you don't want to dispatch it on?

So the `dispatch(req, res)` shouldn't be called when logging in I think.